Fix kernel oops while scanning cciss devices

(This is patches.drivers/cciss-scanning in SLES9_GA_BRANCH.)

diff -burpN linux-2.6.5-7.79.orig/drivers/block/cciss.c linux-2.6.5-7.79/drivers/block/cciss.c
--- linux-2.6.5-7.79.orig/drivers/block/cciss.c	2004-06-16 19:42:14.000000000 -0500
+++ linux-2.6.5-7.79/drivers/block/cciss.c	2004-06-24 13:06:54.214418158 -0500
@@ -773,8 +773,11 @@ static int cciss_ioctl(struct inode *ino
  		luninfo.LunID = drv->LunID;
  		luninfo.num_opens = drv->usage_count;
  		luninfo.num_parts = 0;
- 		/* count partitions 1 to 15 with sizes > 0 */
- 		for(i=1; i <MAX_PART; i++) {
+ 		/* count partitions 0 to 14 with sizes > 0 */
+		/* partition zero to no longer appears to be the entire disk */
+		/* this is a short term hot-fix */
+		/* this ioctl to be replaced with readdir */
+ 		for(i=0; i <MAX_PART-1; i++) {
 			if (!disk->part[i])
 				continue;
 			if (disk->part[i]->nr_sects != 0)
